pkg/client/client.go

```diff
@@ -266,3 +273,38 @@ func buildCustomEndpointsMap(customEndpoints []configv1.AWSServiceEndpoint) map[
 
 	return customEndpointsMap
 }
+
+// sharedCredentialsFileFromSecret returns a location (path) to the shared credentials
+// file that was created using the provided secret
+func sharedCredentialsFileFromSecret(secret *corev1.Secret) (string, error) {
+	var data []byte
+	switch {
+	case len(secret.Data["credentials"]) > 0:
+		data = secret.Data["credentials"]
+	case len(secret.Data["aws_access_key_id"]) > 0 && len(secret.Data["aws_secret_access_key"]) > 0:
+		data = newConfigForStaticCreds(
+			string(secret.Data["aws_access_key_id"]),
+			string(secret.Data["aws_secret_access_key"]),
+		)
+	default:
+		return "", fmt.Errorf("invalid secret for aws credentials")
+	}
+
+	f, err := ioutil.TempFile("", "aws-shared-credentials")
+	if err != nil {
+		return "", fmt.Errorf("failed to create file for shared credentials: %v", err)
+	}
+	defer f.Close()
+	if _, err := f.Write(data); err != nil {
+		return "", fmt.Errorf("failed to write credentials to %s: %v", f.Name(), err)
+	}
+	return f.Name(), nil
```

Comment:
We should return the whole file here (or better, pass in the file object that we opened outside this function?) as it has a method to remove the file for us.

Reply:
Ignore this, it's doing ```defer os.Remove(tmpfile.Name())```

Reply:
Or the method could receive an `io.WriteCloser` so the caller could be the one to handle the filesystem stuff.